src/expressions/refExpression.ts

```diff
@@ -247,7 +261,7 @@ export class RefExpression extends Expression {
     // Check if it needs to be replaced
     if (!this.type || nestDiff > 0 || this.remote !== myRemote) {
```

Comment:
I think this `if` should be `if (!this.type || nestDiff > 0 || this.remote !== myRemote || ignoreCase)` so we actually replace it. Really a test should catch this one.